File: 219-contains-duplicate-ii/contains-duplicate-ii.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {boolean}
 */
var containsNearbyDuplicate = function (nums, k) {
    check = new Map()
    for (let i = 0; i < nums.length; i++) {
        if(check.has(nums[i])){
            if(Math.abs(check.get(nums[i]) - i) <= k){
                return true
            }
        }
        check.set(nums[i],  i)
    }
    return false
};

File: 219-contains-duplicate-ii/contains-duplicate-ii.go
func containsNearbyDuplicate(nums []int, k int) bool {

	checkDistance := map[int]int{}
	for i := 0; i < len(nums); i++ {
		if _, ok := checkDistance[nums[i]]; !ok {
			checkDistance[nums[i]] = i
		} else {
			if i-checkDistance[nums[i]] <= k {
				return true
			}
			checkDistance[nums[i]] = i
		}
	}

	return false
}